### ISSUE TITLE
Create base for new UI library

### ISSUE THREAD

**Issue by markmhendrickson** [blockstack/blockstack-app#3644]
- Research library candidates for React and ReactNative
- Test candidates with POC design set
- Set up new monorepo for shared code across products
- Determine how to distribute tokens via package(s) from new monorepo
- Determine whether to use libraries directly in product repos with theme configs OR wrap into own monorepo for importing
- Choose one library for both React and ReactNative OR two separate ones
- Set up initial theme configurations with current visual elements
- Determine how to replace stacks-ui integration

**Comment by mica000** [comment]
Nice demo of a workflow we could use: https://www.youtube.com/watch?v=7St9GZlJH4k

**Figma tokens -> Framer as storybook -> Chakra UI** 

example of a Figma file with designs tokens: https://www.figma.com/file/rowzp35Jl8qIRapIDjEXFu/Tokens-Studio-for-Figma%3A-Onboarding-(Copy)?type=design&node-id=0%3A1&t=TGQRPoERVnRGtT4m-1

Example of a source of truth made with Framer: https://framer.design/

**Comment by kyranjamie** [comment]
Thanks for sharing Michelly. I've only skim watched this video, but Figma tokens looks great. Totally up for using this.

I don't know if I'd consider the Framer site as an alternative to Storybook. It might make a great brand colophon kinda site, but wouldn't work well as the home for our UI components, which is where we'd probably colocate Storybook. Open for trying Framwer React component export, but very likely has limitations.

This approach should work well with any themeable ui library.

Current thinking:

```mermaid
flowchart TB
  A(FigmaTokens) --> B("Figma brand site/wallet website")
  A --Auto-updated--> C("@wallet/tokens")
  C --Mapping tokens to theme--> D(Web-based UI library)
  C --Mapping tokens to theme--> E(React Native-based UI library)
```

**Comment by pete-watters** [comment]
Thanks for sharing the video Michelly and for the diagram and ideas above Kyran. I watched the video today and it does look really interesting. I spent some time today researching this and library candidates and some of my notes/ feedback is

**Figma tokens**
I think that Figma tokens will be really useful and we should use them for a lot of things: colours, spacing, typography etc.

**Framer**
I think that Framer is great for marketing/websites also but I believe for us it would be better to use Storybook for our app components as then we can more easily add tests and maintain a coding playground. Storybook isn't perfect and there maybe a better alternative but I think it ticks most boxes and is pretty standard

**Chakra / Panda etc.**
- Chakra seems to be making a lot of improvements - changing to split things up and be multi platform (Panda, Zag)
- They still don't offer native support though but they said they will do it  '[pretty soon](https://twitter.com/chakra_ui/status/1388033321213612032?s=21)' in 2021  
- A lot of people have been [pushing for React Native support ](https://github.com/chakra-ui/chakra-ui/issues/183)

There is an interesting article here about the [future of chakra](https://www.adebayosegun.com/blog/the-future-of-chakra-ui ). If we were just creating a web only platform or choose to have different libraries for web and native then I'd be all for using Panda + Zag.js.

**Library for React + React Native**
As we are changing the UI library now and plan to create a mobile app soon, I think it would be prudent to consider choosing a library that will support both. From my initial research there seems to be a few options out there but the most promising seem to be:
- [**Nativebase** ](https://nativebase.io/ )
    - Native base supposedly has performance issues 
    - https://dev.to/redbar0n/how-to-use-chakraui-on-native-mobile-53hg 
    - https://github.com/GeekyAnts/NativeBase/issues/4302 
    - It has a lot of app development starter kits though

- [**Tamagui**](https://tamagui.dev/ )
    - Looks promising and seems to get better feedback from people pushing for Chakra on native
    - They offer a paid [TakeOut starter](https://tamagui.dev/takeout) also which is intriguing

**Web Extension support Q**
I'm adding on a Q about web extension support here too. I think we support chromium and webkit browsers but not Edge and Safari if I'm not mistaken? Is there a reason we can't support this? 

Asking as if we are changing somethings now, perhaps its a good time to also consider trying to support more browsers. Maybe using Next.js would help us to achieve that?

**Comment by pete-watters** [comment]
We are working on this as part of the mono-repo and will add storybook for web and react-native there

https://github.com/leather-wallet/extension/issues/1230

**Comment by markmhendrickson** [comment]
It seems we can close this particular issue since we're rolled out the new library. @pete-watters agreed?

**Comment by pete-watters** [comment]
Yes lets close this and continue work here https://github.com/leather-wallet/extension/issues/1230. We haven't actually implemented this but we will in #1230
